erdpy/config.py

```diff
@@ -162,10 +165,56 @@ def get_defaults() -> Dict[str, Any]:
 
 
 def read_file() -> Dict[str, Any]:
-    if not os.path.isfile(CONFIG_PATH):
-        return dict()
-    return utils.read_json_file(CONFIG_PATH)
+    if os.path.isfile(LOCAL_CONFIG_PATH):
+        return utils.read_json_file(LOCAL_CONFIG_PATH)
+    if os.path.isfile(CONFIG_PATH):
+        return utils.read_json_file(CONFIG_PATH)
+    return dict()
 
 
 def write_file(data: Dict[str, Any]):
-    utils.write_json_file(CONFIG_PATH, data)
+    if os.path.isfile(LOCAL_CONFIG_PATH):
+        utils.write_json_file(LOCAL_CONFIG_PATH, data)
+    else:
+        utils.write_json_file(CONFIG_PATH, data)
+
+
+def add_config_args(argv):
+    try:
+        func, subcommand, *_ = argv
+    except ValueError:
+        return argv
+
+    config = read_file()
+
+    try:
+        config_args = config[func][subcommand]
+    except KeyError:
+        return argv
+
+    extra_args = []
+    for key, value in config_args.items():
+        key_arg = f'--{key}'
+        # arguments from the command line override the config
+        if key_arg in argv:
+            continue
+        extra_args.append(key_arg)
+        if value is True:
+            continue
+        if isinstance(value, List):
+            for item in value:
+                extra_args.append(str(item))
+        else:
+            extra_args.append(str(value))
```

Comment:
A separate function called `determine_extra_args()` (or another name) could contain this code.

Reply:
Done. Created a `determine_final_args` function.